R/weight.R

```diff
@@ -41,18 +43,73 @@ weight <- function (x) {
     currentWeight <- ShojiEntity(crGET(shojiURL(x, "fragments", "preferences")))$weight
     if (!identical(value, currentWeight)) {
         crPATCH(shojiURL(x, "fragments", "preferences"),
-            body=toJSON(list(weight=value)))
+            body = toJSON(list(weight = value)))
         x <- refresh(x)
     }
     return(x)
 }
 
 #' @rdname weight
 #' @export
+is.weight <- function (x) {
+    if (is.variable(x)) {
+        ds <- loadDataset(datasetReference(x))
+        return(identical(weight(ds), x))
+    } else {
+        return(FALSE)
+    }
+}
+
+#' @rdname weight
+#' @export
+setMethod("is.weight<-", "NumericVariable", function (x, value) {
+    isTRUEorFALSE(value)
+    ds <- loadDataset(datasetReference(x))
+    if (value) {
+        weight(ds) <- x
+    } else {
+        weight(ds) <- NULL
+    }
+    return(x)
+})
+
+#' Change which variables can be set as weights
+#'
+#' Editors can change which variables can be set as the weighting variable
+#' for a dataset. For instance if several weights have been calculated they
+#' can let the user choose which of those variables to use a weight, but prevent
+#' the user from choosing other variables as weight. This function allows you
+#' to get and set the `weightVariables` of a dataset, which are the variables which
+#' can be set as the dataset [weight()].
+#'
+#' @param x a CrunchDataset
+#' @param value For the setter variables to set as the `weightVariables` this can
+#' be a numeric Crunch variable, list of numeric Crunch variables or
+#' a character vector with the aliases of numeric Crunch variables.
+#'
+#' @return For the getter the variables which can be set as the dataset's weight. For
+#' the setter the dataset duly modified.
+#' @seealso [weight()] [makeWeight()]
```

Comment:
Perhaps see also `modifyWeightVariables`, which gives you greater control. In fact, consider documenting all of these together and indicate that `weightVariables<-` is a convenience method around `modifyWeightVariables(mode="append")`